### PR TITLE
Select correct access group for SubscriptionTokenKeychainStorage

### PR DESCRIPTION
**Pull request by diegoreymendez**
Task/Issue URL: https://app.asana.com/0/547792610048271/1207483115895951/f
Tech Design URL:
CC:

## Description

We're using .unspecified access group for SubscriptionTokenKeychainStorage .  While this is not causing trouble, it's not correct and should be changed.

## Testing

Make sure the VPN works and connects.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/AppDependencyProvider.swift [hunk 1]
```diff
@@ -134,13 +134,11 @@ class AppDependencyProvider: DependencyProvider {
             purchasePlatform: .appStore)
         let accessTokenProvider: () -> String? = {
             func isSubscriptionEnabled() -> Bool {
-                if let subscriptionOverrideEnabled = UserDefaults.networkProtectionGroupDefaults.subscriptionOverrideEnabled {
 #if ALPHA || DEBUG
+                if let subscriptionOverrideEnabled = UserDefaults.networkProtectionGroupDefaults.subscriptionOverrideEnabled {
                     return subscriptionOverrideEnabled
-#else
-                    return false
```
Comment (diegoreymendez) [inline DuckDuckGo/AppDependencyProvider.swift:141]: Release users shouldn't ever go in here, so let's just clean things up so they won't be allowed to.

File: PacketTunnelProvider/NetworkProtection/NetworkProtectionPacketTunnelProvider.swift [hunk 2]
```diff
@@ -272,7 +272,9 @@ final class NetworkProtectionPacketTunnelProvider: PacketTunnelProvider {
         let entitlementsCache = UserDefaultsCache<[Entitlement]>(userDefaults: UserDefaults.standard,
                                                                  key: UserDefaultsCacheKey.subscriptionEntitlements,
                                                                  settings: UserDefaultsCacheSettings(defaultExpirationInterval: .minutes(20)))
-        let accessTokenStorage = SubscriptionTokenKeychainStorage(keychainType: .dataProtection(.unspecified))
```
Comment (diegoreymendez) [inline PacketTunnelProvider/NetworkProtection/NetworkProtectionPacketTunnelProvider.swift:275]: Unspecified isn't right.

**Review by miasma13 — APPROVED** [review]
👍

**Review by samsymons — APPROVED** [review]

**Comment by diegoreymendez** [comment]
Task/Issue URL: https://app.asana.com/0/547792610048271/1207483115895951/f
Tech Design URL:
CC:

## Description

We're using .unspecified access group for SubscriptionTokenKeychainStorage .  While this is not causing trouble, it's not correct and should be changed.

**Pull request merged**